File: 7_function_overloading.py
class Calculator:
    def devide(self,*args):
        if len(args)==2:
            return args[0]/args[1]
        elif len(args)==3:
            return (args[0]/args[1])/args[2]
        elif len(args)==4:
            return ((args[0]/args[1])/(args[2]/args[3]))


c = Calculator()
print(c.devide(20,2))
print(c.devide(300,2,3))
print(c.devide(500,4,3,2))

File: 7_function_overloading.cpp
#include <bits/stdc++.h>
#include<iostream>
#include <string>
using namespace std;
class Calculator{
    // int x,y,z;
    // Calculator(int x,int y,int z){
    //     this->x=x;
    //     this->y=y;
    //     this->z=z;
    // }
    public:
        int devide(int a,int b,int c){
            return (a/b)/c;
        }
        float devide(int a,int b){
            return a/b;
        }
        double devide(double a,double b){
            return a/b;
        }
};


int main(){
    Calculator c;
    int d = c.devide(100,2,2);
    float e = c.devide(99.3,2.4);
    double f = c.devide(100,5);
    cout<<"d-> "<<d<<endl;
    cout<<"e-> "<<e<<endl;
    cout<<"f-> "<<f<<endl;
}
